### DISCUSSION TITLE
How to invoke official datatracker api search endpoint

### DISCUSSION THREAD

**Discussion by fgtrzah** [ietf-tools/datatracker#7432] [Help / Questions]
Where can I find documentation for the api endpoint that the following datatracker search feature uses to conduct search? (does one exist?)

![Screenshot 2024-05-17 at 12 02 30 PM](https://github.com/ietf-tools/datatracker/assets/118348456/5ccbc3ee-49b1-4430-9ad8-d5e82bccd25d)

I didn't see any graphql or rest api endpoint appear in network tab when using the search page. I'm beautiful soup and bespoke scraping to programmatically search datatracker for one of my projects, I'd rather not if the datatracker api has a search endpoint but according to:

[/api/v1](https://datatracker.ietf.org/api/v1)

There isn't one? Details of my current solution outlined [here](https://github.com/fgtrzah/llmrfcpoc/blob/main/rfcllm/search/routes.py) see the `search_ietf` function for reference. 

Also, didn't see any discussion related to search api:

![Screenshot 2024-05-17 at 12 11 35 PM](https://github.com/ietf-tools/datatracker/assets/118348456/403e7058-5adc-426c-91a2-4867bdf2602d)

Please let me know if there's more appropriate place to ask help for this dont know if this belongs here or in issues.

**Comment by rjsparks** [comment]
No there isn't currently a specific search api endpoint at the datatracker.

The v1 api only exposes the django models - you could _build_ search using it, but it requires understanding the models. (You've already started to build some simple ones at https://github.com/fgtrzah/llmrfcpoc/blob/905534659a3c9302c2a393d85952f79a2e855b81/rfcllm/search/routes.py#L52).

The html search endpoint is also fairly limited - be sure it can do what you really need before trying to scrape it.

What _kind_ of searches are you wanting to perform?

**Reply by fgtrzah** [reply]
I'd like to perform semantic search using vectors, but for now I'm just looking to move away from a scraper and utilize said django models to build a rest api endpoint that can query against the title, author, and description. Is there a repository that houses the models? [e.g. is this endpoint describing the models?](https://datatracker.ietf.org/api/v1/dbtemplate/dbtemplate/?limit=2000&format=json) If the IETF has any conventions around how RFC documents *should* be searched then I'd like to abide by those in my implementation. 

Lastly, if scraping violates any policies where should I go to find out more (specifically about scraping + properly giving credit the the datatracker platform/org)?
